src/main/java/seedu/address/logic/commands/PoolCommand.java

```diff
@@ -76,38 +96,25 @@ public CommandResult execute(Model model) throws CommandException {
             }
         }
 
+        // obtain passengers from indices
+        Set<Passenger> passengersToPool = new HashSet<>();
+
         for (Index idx : passengers) {
-            Passenger passengerToEdit = lastShownList.get(idx.getZeroBased());
-            Passenger editedPassenger = assignDriverToPassenger(passengerToEdit, driver);
-            joiner.add(editedPassenger.getName().toString());
-            model.setPassenger(passengerToEdit, editedPassenger);
+            Passenger passenger = lastShownList.get(idx.getZeroBased());
+            assert passenger != null : "passenger should not be null";
+            passengersToPool.add(passenger);
         }
 
-        model.updateFilteredPassengerList(PREDICATE_SHOW_ALL_PASSENGERS);
-
-        return new CommandResult(String.format(MESSAGE_POOL_SUCCESS, driver.toString(), joiner.toString()));
-    }
+        Pool toAdd = new Pool(driver, tripDay, tripTime, passengersToPool, tags);
 
-    /**
-     * Assigns the given {@code Driver} to the given {@code Passenger}.
-     * @param passengerToEdit the {@code Passenger} to add the {@code Driver} to.
-     * @param driver the {@code Driver} to add to the {@code Passenger}.
-     * @return a new {@code Passenger}, with the given driver assigned.
-     */
-    private static Passenger assignDriverToPassenger(Passenger passengerToEdit, Driver driver) {
-        requireNonNull(passengerToEdit);
-        requireNonNull(driver);
+        if (model.hasPool(toAdd)) {
+            throw new CommandException(MESSAGE_DUPLICATE_POOL);
+        }
 
-        Name updatedName = passengerToEdit.getName();
-        Phone updatedPhone = passengerToEdit.getPhone();
-        Address updatedAddress = passengerToEdit.getAddress();
-        Set<Tag> updatedTags = passengerToEdit.getTags();
-        TripDay updatedTripDay = passengerToEdit.getTripDay();
-        TripTime updatedTripTime = passengerToEdit.getTripTime();
-        Optional<Price> price = passengerToEdit.getPrice();
+        model.addPool(toAdd);
+        model.updateFilteredPassengerList(PREDICATE_SHOW_ALL);
```

Comment:
Is it necessary to update the passenger list after adding the pool? As far as I understand there shouldn't be a change in the passenger list after pooling?

Reply:
rectified